controllers/testclustertopology_controller.go

```diff
@@ -115,7 +115,52 @@ func (r *TestClusterTopologyReconciler) Reconcile(req ctrl.Request) (ctrl.Result
 			// TODO: wait for the cluster be ready
 		}
 	case naglfarv1.ClusterTopologyStateReady:
-		// DO NOTHING
+		// first create
+		if ct.Status.PreServerConfigs == nil && ct.Status.PreVersion == nil {
+			ct.Status.PreServerConfigs = ct.Spec.TiDBCluster.ServerConfigs.DeepCopy()
+			ct.Status.PreVersion = ct.Spec.TiDBCluster.Version.DeepCopy()
+			if err := r.Status().Update(ctx, &ct); err != nil {
+				log.Error(err, "unable to update TestClusterTopology")
+				return ctrl.Result{RequeueAfter: 10 * time.Second}, err
```

Comment:
You should select one of returning an error or re-queue.

Reply:
done

Reply:
change: ctrl.Result{},err